### ISSUE TITLE
Supporting customization of float encoding in JSON

### ISSUE THREAD

**Issue by 717908ce-bb9c-4e05-b89c-d98143e51333** [python/cpython#81022]
BPO | [36841](https://bugs.python.org/issue36841)
--- | :---
Nosy | @rhettinger, @mdickinson, @ezio-melotti, @merwok, @mitar, @minrk, @Lee-W
PRs | <li>python/cpython#13233</li><li>python/cpython#28648</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-05-07.20:32:33.357>
labels = ['type-feature', 'library', '3.11']
title = 'Supporting customization of float encoding in JSON'
updated_at = <Date 2021-09-30.12:47:51.360>
user = 'https://github.com/mitar'
```

bugs.python.org fields:
```python
activity = <Date 2021-09-30.12:47:51.360>
actor = 'mark.dickinson'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2019-05-07.20:32:33.357>
creator = 'mitar'
dependencies = []
files = []
hgrepos = []
issue_num = 36841
keywords = ['patch']
message_count = 6.0
messages = ['341817', '341904', '341914', '342350', '393441', '402950']
nosy_count = 7.0
nosy_names = ['rhettinger', 'mark.dickinson', 'ezio.melotti', 'eric.araujo', 'mitar', 'minrk', 'clleew']
pr_nums = ['13233', '28648']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue36841'
versions = ['Python 3.11']
```

</p></details>

**Comment by arhadthedev** [comment]
@Lee-W (as an author of a child PR) The Discourse is https://discuss.python.org/.

**Comment by mdickinson** [comment]
The existing thread https://discuss.python.org/t/harmonizing-the-json-serialization-of-non-finite-float-values-with-browsers-and-other-languages/20477 may be relevant.

**Comment by oren-hecht** [comment]
Is there any updates on this issue? 

the fact that the standard `json` library is incompatible with pretty much all default Javascript parsers seems like a pretty big deal - even if it is only with non-finite float values..

I think the solution proposed [in this thread](https://discuss.python.org/t/harmonizing-the-json-serialization-of-non-finite-float-values-with-browsers-and-other-languages/20477) can definitely be a decent solution!

**Comment by mdickinson** [comment]
Yes, @encukou's second option in [this message](https://discuss.python.org/t/harmonizing-the-json-serialization-of-non-finite-float-values-with-browsers-and-other-languages/20477/4) still seems like a good resolution to me. That is:

- allow the existing `allow_nan` parameter to take the value `"null"` (and in that case, convert both infinities and NaNs to JSON `null`)
- also deprecate non-boolean, non-`"null"` values for `allow_nan`

I'm not likely to have cycles to work on this any time soon, but would be willing to review a PR.

**Comment by mdickinson** [comment]
See https://github.com/python/cpython/pull/115246.
